filer/management/commands/filer_check.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import os
+
+from django.core.files.storage import DefaultStorage
+from django.core.management.base import BaseCommand, CommandError
+from django.utils.module_loading import import_string
+from django.utils.six.moves import input
+from django.utils.translation import ugettext_lazy as _
+from filer.settings import DEFAULT_FILER_STORAGES
+
+
+class Command(BaseCommand):
+    help = _("Look for orphaned files in media folders.")
+    storage = DefaultStorage()
+    prefix = DEFAULT_FILER_STORAGES['public']['main']['UPLOAD_TO_PREFIX']
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--orphans',
+            action='store_true',
+            dest='orphans',
+            default=False,
+            help=_("Walk through the media folders and look for orphaned files."),
+        )
+        parser.add_argument(
+            '--delete-orphans',
+            action='store_true',
+            dest='delete_orphans',
+            default=False,
+            help=_("Delete orphaned files from their media folders."),
+        )
+        parser.add_argument(
+            '--missing',
+            action='store_true',
+            dest='missing',
+            default=False,
+            help=_("Verify media folders and report about missing files."),
+        )
+        parser.add_argument(
+            '--delete-missing',
+            action='store_true',
+            dest='delete_missing',
+            default=False,
+            help=_("Delete references in database if files are missing in media folder."),
+        )
+        parser.add_argument(
+            '--noinput',
+            '--no-input',
+            action='store_false',
+            dest='interactive',
+            default=True,
+            help="Do NOT prompt the user for input of any kind."
+        )
+
+    def handle(self, *args, **options):
+        if options['missing']:
+            self.verify_references(options)
+        if options['delete_missing']:
+            if options['interactive']:
+                msg = _("\nThis will delete entries from your database. Are you sure you want to do this?\n\n"
+                        "Type 'yes' to continue, or 'no' to cancel: ")
+                if input(msg) != 'yes':
+                    raise CommandError("Aborted: Delete missing file entries from database.")
```

Comment:
why raise an exception, no is a valid option

Reply:
fixed